Find a functionally equivalent code:android.os.NullVibrator.vibrate:COMMENT
Method Modifier: public      hidden      
Comment:/**
 * @hide
 */

@@ -1,3 +1,5 @@
 {
-    vibrate(pattern, repeat);
+    if (repeat >= pattern.length) {
+        throw new ArrayIndexOutOfBoundsException();
+    }
 }

